client/plots/brainImaging.js

```diff
@@ -246,154 +220,98 @@ class BrainImaging {
-				const img = this.dom.tdL.append('div').append('img').attr('src', result.url)
-				this.dom.imagesL.push(img)
-			}
+		this.imagesData = {
+			brainImageL: { dataUrls: {}, td: this.dom.tdL },
+			brainImageF: { dataUrls: {}, td: this.dom.tdF },
+			brainImageT: { dataUrls: {}, td: this.dom.tdT }
 		}
 
-		if (this.state.updateF) {
-			const body = {
-				genome: this.state.genome,
-				dslabel: this.state.dslabel,
-				refKey: this.state.config.queryKey,
-				f: this.settings.brainImageF,
-				selectedSampleFileNames: this.state.config.selectedSampleFileNames,
-				divideByTW,
-				overlayTW,
-				legendFilter: this.state.config.legendFilter
-			}
-			const data = await dofetch3('brainImaging', { body })
-			if (data.error) throw data.error
+		this.datas = await this.requestImages(Object.keys(this.imagesData))
+		this.imagesData.brainImageL.data = this.datas[0]
+		this.imagesData.brainImageF.data = this.datas[1]
+		this.imagesData.brainImageT.data = this.datas[2]
+		for (const key in this.imagesData) this.renderImages(key)
 
-			this.dataUrlF = []
-			for (const [termV, result] of Object.entries(data.brainImage)) {
-				this.dataUrlF.push(result)
-			}
-			this.dom.tdF.selectAll('*').remove()
-			this.dom.imagesF = []
-			for (const [termV, result] of Object.entries(this.dataUrlF)) {
-				if (divideByTW)
-					this.dom.tdF
-						.append('div')
-						.attr('class', 'pp-chart-title')
-						.style('text-align', 'center')
-						.html('&nbsp;')
-						.style('font-weight', '600')
-						.style('font-size', '24px')
-						.style('margin-bottom', '5px')
-						.style('margin-top', '5px')
-						.style('display', 'block')
-				const img = this.dom.tdF.append('div').append('img').attr('src', result.url)
-				this.dom.imagesF.push(img)
-			}
+		this.renderLegend()
+	}
+
+	renderImages(key) {
+		const data = this.imagesData[key].data
+		const td = this.imagesData[key].td
+		this.legendValues = data.legend
+		if (data.error) throw data.error
+		const dataUrls = this.imagesData[key].dataUrls
+		for (const [termV, result] of Object.entries(data.brainImage)) {
+			dataUrls[termV] = result
 		}
-		if (this.state.updateT) {
+
+		td.selectAll('*').remove()
+		for (const [termV, result] of Object.entries(dataUrls)) {
+			if (this.state.config.divideByTW)
+				td.append('div')
+					.attr('class', 'pp-chart-title')
+					.style('text-align', 'center')
+					.text(`${termV} (n=${result.catNum})`)
+					.style('font-weight', '600')
+					.style('color', 'white')
+					.style('font-size', '24px')
+					.style('margin-bottom', '5px')
+					.style('margin-top', '5px')
+					.style('display', 'block')
+			td.append('div').append('img').attr('src', result.url)
+		}
+	}
+
+	async requestImages(keys) {
+		const promises = []
+		for (const key of keys) {
 			const body = {
 				genome: this.state.genome,
 				dslabel: this.state.dslabel,
 				refKey: this.state.config.queryKey,
-				t: this.settings.brainImageT,
+				l: this.settings[key],
```

Comment:
I think `l` should be `f` or `t`, depending on the key? Instead of just passing array of string `keys` as argument, can pass an array of object that includes this single letter key as a property, 
```js
...
this.imagesData = {
	brainImageL: { dataUrls: {}, td: this.dom.tdL, key: 'l', value: this.settings.brainImageL },
	brainImageF: { dataUrls: {}, td: this.dom.tdF, key: 'f', value: this.settings.brainImageF },
	brainImageT: { dataUrls: {}, td: this.dom.tdT, key: 't', value: this.settings.brainImageT }
}
...
this.datas = await this.requestImages(Object.values(this.imagesData))
...
async requestImages(images) {
    const promises = []
		for (const img of images) {
			const body = {
				genome: this.state.genome,
				dslabel: this.state.dslabel,
				refKey: this.state.config.queryKey,
				[img.key]: [img.value],
                         }  
}
```